File: Leetcode/Array/#448. Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
    for(int i=0; i<nums.size(); i++){
        while(nums[nums[i]-1]!=nums[i]) swap(nums[nums[i]-1],nums[i]);
    }
    vector<int>res;
        for(int i=0; i<nums.size(); i++){
            if(i+1!=nums[i]) res.push_back(i+1);
        }
        return res;
    }
};

//1 2 2 3 3 4 7 8 8

File: Array/#448. Find All Numbers Disappeared in an Array.py
'''
TC: O(n)
SC: O(1)
'''

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []
        for e in nums:
            e = abs(e)
            nums[e - 1] = -1 * abs(nums[e - 1])

        for i in range(len(nums)):
            if nums[i] > 0:
                res.append(i + 1)

        return res